lib/CdeContext.tsx

```diff
@@ -1,83 +1,159 @@
+} from "./services/validatorsService.ts";
+import {mapDatasetSample, mapStringTableToDatasetMapping} from "./services/initialMappingService.ts";
+import {updateDatasetMappingRow} from "./services/updateMappingService.ts";
 
 export const CdeContext = createContext<{
-    labName: string;
-    config: Config;
-    inputMappings: InputMapping[][];
-    mapping: DatasetCDEMapping;
-    setMapping: (mapping: DatasetCDEMapping) => void;
+
+    name: string;
+    datasetSample: DatasetSample;
+    datasetMapping: DatasetMapping;
+    datasetMappingHeader: string[];
+    collections: Collection[]
+    config: Config
+
+
+    // UI Module
     step: number;
     setStep: (step: number) => void;
     loadingMessage: string | null;
     setLoadingMessage: (loadingMessage: string | null) => void;
     errorMessage: string | null;
     setErrorMessage: (errorMessage: string | null) => void;
-    isOpen: boolean;
     handleClose: () => void;
-    infoOpen: boolean;
-    setInfoOpen: (open: boolean) => void;
 }>({
-    labName: '',
+
+    name: '',
+    datasetSample: [],
+    datasetMapping: {},
+    datasetMappingHeader: [],
+    collections: [],
     config: {
         width: "100%",
         height: "100%",
     },
-    inputMappings: [],
-    mapping: {} as DatasetCDEMapping,
-    setMapping: () => {},
+
     step: 0,
-    setStep: () => {},
+    setStep: () => {
+    },
     loadingMessage: null,
-    setLoadingMessage: () => {},
+    setLoadingMessage: () => {
+    },
     errorMessage: null,
-    setErrorMessage: () => {},
-    isOpen: true,
-    handleClose: () => { },
-    infoOpen: false,
-    setInfoOpen: () => {}
+    setErrorMessage: () => {
+    },
+    handleClose: () => {
+    },
 });
 
 export const useCdeContext = () => useContext(CdeContext);
 
-export const CdeContextProvider = ({children, labName, callback, inputMappings, config}: PropsWithChildren<InitParams>) => {
-    const [mapping, setMapping] = useState<DatasetCDEMapping>({});
-    const [step, setStep] = useState<number>(STEPS.HOME);
-    const [loadingMessage, setLoadingMessage] = useState<string | null>(null);
-    const [errorMessage, setErrorMessage] = useState<string | null>(null)
-    const [isOpen, setIsOpen] = useState<boolean>(true)
-    const [infoOpen, setInfoOpen] = useState<boolean>(false)
+export const CdeContextProvider = ({
+                                       datasetSample: rawDatasetSample,
+                                       datasetMapping: rawDatasetMapping,
+                                       additionalDatasetMappings: rawAdditionalDatasetMappings,
+                                       collections,
+                                       config,
+                                       name,
+                                       callback,
+                                       children
+                                   }: PropsWithChildren<InitParams>) => {
+
+    let datasetSample;
+    // Process and validate datasetSample
```

Comment:
Moves the validation and mapping from Home to the Context creation as discussed in a meeting.